failure-summary.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Finds failing Checks and combines each help message into an array
+ * @param  {Object} nodeData Individual "detail" object to generate help messages for
+ * @return {String}          failure messages
+ */
+var failureSummary = function failureSummary(nodeData) {
+	'use strict';
+
+	var failingChecks = {};
+	// combine "all" and "none" as messaging is the same
+	failingChecks.none = nodeData.none.concat(nodeData.all);
+	failingChecks.any = nodeData.any;
+
+	return Object.keys(failingChecks).map(function (key) {
```

Comment:
There should be a message in front of each grouping to instruct the user how to interpret the output. In the case of `any` this message should be along the lines of "fix any of the following" and for the `none` group it should be "fix all of the following".